geode-core/src/distributedTest/java/org/apache/geode/cache/query/internal/QueryConfigurationServiceConstraintsDistributedTest.java

```diff
@@ -0,0 +1,595 @@
+import org.apache.geode.test.junit.categories.OQLQueryTest;
+import org.apache.geode.test.junit.categories.SecurityTest;
+import org.apache.geode.test.junit.rules.serializable.SerializableTemporaryFolder;
+import org.apache.geode.test.junit.rules.serializable.SerializableTestName;
+
+@RunWith(JUnitParamsRunner.class)
+@Category({OQLQueryTest.class, SecurityTest.class})
+public class QueryConfigurationServiceConstraintsDistributedTest implements Serializable {
+  private static final int ENTRIES = 500;
+  private static final int PUT_KEY = ENTRIES + 1;
+  private static final int CREATE_KEY = ENTRIES + 2;
+  private static final int REMOVE_KEY = ENTRIES - 1;
+  private static final int DESTROY_KEY = ENTRIES - 2;
+  private static final int UPDATE_KEY = ENTRIES - 3;
+  private static final int REPLACE_KEY = ENTRIES - 4;
+  private static final int INVALIDATE_KEY = ENTRIES - 5;
+  private File logFile;
+  protected MemberVM server;
+  protected ClientVM client;
+
+  @Rule
+  public ClusterStartupRule cluster = new ClusterStartupRule();
+
+  @Rule
+  public SerializableTestName testName = new SerializableTestName();
+
+  @Rule
+  public SerializableTemporaryFolder temporaryFolder = new SerializableTemporaryFolder();
+
+  @SuppressWarnings("unused")
+  private Object[] getRegionTypeAndOperation() {
+    return new Object[] {
+        new Object[] {"LOCAL", "PUT"},
+        new Object[] {"LOCAL", "CREATE"},
+        new Object[] {"LOCAL", "REMOVE"},
+        new Object[] {"LOCAL", "DESTROY"},
+        new Object[] {"LOCAL", "UPDATE"},
+        new Object[] {"LOCAL", "REPLACE"},
+        new Object[] {"LOCAL", "INVALIDATE"},
+
+        new Object[] {"REPLICATE", "PUT"},
+        new Object[] {"REPLICATE", "CREATE"},
+        new Object[] {"REPLICATE", "REMOVE"},
+        new Object[] {"REPLICATE", "DESTROY"},
+        new Object[] {"REPLICATE", "UPDATE"},
+        new Object[] {"REPLICATE", "REPLACE"},
+        new Object[] {"REPLICATE", "INVALIDATE"},
+
+        new Object[] {"PARTITION", "PUT"},
+        new Object[] {"PARTITION", "CREATE"},
+        new Object[] {"PARTITION", "REMOVE"},
+        new Object[] {"PARTITION", "DESTROY"},
+        new Object[] {"PARTITION", "UPDATE"},
+        new Object[] {"PARTITION", "REPLACE"},
+        new Object[] {"PARTITION", "INVALIDATE"},
+    };
+  }
+
+  @Before
+  public void setUp() throws Exception {
+    logFile = temporaryFolder.newFile(testName.getMethodName());
+
+    server = cluster.startServerVM(1, cf -> cf
+        .withProperty(SECURITY_MANAGER, SimpleSecurityManager.class.getName())
+        .withProperty(SERIALIZABLE_OBJECT_FILTER, "org.apache.geode.cache.query.internal.*")
+        .withProperty("security-username", "cluster").withProperty("security-password", "cluster")
+        .withProperty("log-file", logFile.getAbsolutePath()));
+
+    server.invoke(() -> {
+      assertThat(ClusterStartupRule.getCache()).isNotNull();
+      InternalCache internalCache = ClusterStartupRule.getCache();
```

Comment:
Instead of two method calls here, it might be more efficient to assign the `internalCache` variable first and then assert that the variable is not null. This applies to almost every time `ClusterStartupRule` is used to retrieve a cache in this file and the other file in this PR.

Reply:
I didn't pay much attention when originally doing this as we're talking about a test class here, not sure how much more "efficient" avoiding a method call would be, and how important it actually is for the test, but I made the requested changed anyway 👍